core/controllers/reader.py

```diff
@@ -1040,3 +1040,34 @@ def get(self):
             'question_dicts': question_dicts
         })
         self.render_json(self.values)
+
+
+class LearnerAnswerDetailsSubmissionHandler(base.BaseHandler):
+    """Handles the learner answer details submission."""
+
+    @acl_decorators.can_submit_answer_details
+    def put(self, entity_id, entity_type):
+        """"Handles the PUT requests. Stores the answer details submitted
+        by the learner.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        interaction_id = self.payload.get('interaction_id')
+        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
```

Comment:
Add elif question case; add else case

Reply:
Also currently there is no ACL layer for can_play_question, should I do it after the implementation of the project for the exploration?

Reply:
Make everything work for both explorations and questions in this PR.

Reply:
Done